src/databricks/labs/ucx/hive_metastore/tables.py

```diff
@@ -475,17 +475,21 @@ def _describe(self, catalog: str, database: str, table: str) -> Table | None:
 
 
 class FasterTableScanCrawler(CrawlerBase):
-    def _try_fetch(self) -> Iterable[Table]:
-        """Tries to load table information from the database or throws TABLE_OR_VIEW_NOT_FOUND error"""
-        for row in self._fetch(f"SELECT * FROM {escape_sql_identifier(self.full_name)}"):
-            yield Table(*row)
+    """
+    FasterTableScanCrawler is a specialized version of TablesCrawler that uses spark._jsparkSession to utilize
+    faster scanning with Scala APIs.
+
+    For integration testing, FasterTableScanCrawler is tested using the larger assessment test rather than
+    just the class. Testing the class individually would require utilizing a remote spark connection with the
+    Databricks workspace.
+    """
 
     def __init__(self, backend: SqlBackend, schema, include_databases: list[str] | None = None):
         self._backend = backend
         self._include_database = include_databases
 
         # pylint: disable-next=import-error,import-outside-toplevel
-        from pyspark.sql.session import SparkSession  # type: ignore[import-not-found]
+        from pyspark.sql.session import SparkSession
 
         super().__init__(backend, "hive_metastore", schema, "tables", Table)
         self._spark = SparkSession.builder.getOrCreate()
```

Comment:
Can we move this onto a cached property?

Something like:
```python
@cached_property
def _spark(self):
    # pylint: disable-next=import-error,import-outside-toplevel
    from pyspark.sql.session import SparkSession # type: ignore[import-not-found]

    return SparkSession.builder.getOrCreate()

Reply:
it's still initialised only once per job task, not sure if it's necessary.